From ea3ec9031a41779826b521d4a7a7b1a71ebeebf8 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Fri, 12 Apr 2019 16:15:36 +0530
Subject: [PATCH 145/386] octeontx2-pf: Use sleepable mbox apis wherever
 possible

Busy polling for mailbox responses is not required unless
sending messages are guarded by spin_locks. Since spin_locks
are not used for PTP messages busy polling is removed.

Change-Id: Ic093e3600dbab4f323913cfa1a61b43498763eb1
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/7303
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_ptp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ptp.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ptp.c
index caa5266132f6..4211930746fd 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ptp.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ptp.c
@@ -39,7 +39,7 @@ static int otx2_ptp_adjfine(struct ptp_clock_info *ptp_info, long scaled_ppm)
 	req->op = PTP_OP_ADJFINE;
 	req->scaled_ppm = scaled_ppm;
 
-	err = otx2_sync_mbox_msg_busy_poll(&ptp->nic->mbox);
+	err = otx2_sync_mbox_msg(&ptp->nic->mbox);
 	if (err)
 		return err;
 
@@ -62,7 +62,7 @@ static u64 ptp_cc_read(const struct cyclecounter *cc)
 
 	req->op = PTP_OP_GET_CLOCK;
 
-	err = otx2_sync_mbox_msg_busy_poll(&ptp->nic->mbox);
+	err = otx2_sync_mbox_msg(&ptp->nic->mbox);
 	if (err)
 		return 0;
 
-- 
2.17.1

